eth_typing/abi.py

```diff
@@ -17,7 +21,38 @@
 """Binary data to be decoded."""
 
 
-class ABIComponent(TypedDict, total=False):
+@deprecated("`ABIEventComponent` is deprected, use `ABIComponent` instead.")
```

Comment:
```suggestion
@deprecated("`ABIEventComponent` is deprecated, use `ABIComponent` instead.")
```

Reply:
Since there's a common message, we could factor it out:

```py
ABI_COMPONENT_DEPRECATION_MSG = "`{0}` is deprecated, use `ABIComponent` instead."

@deprecated(ABI_COMPONENT_DEPRECATION_MSG.format("ABIEventComponent"))
...

@deprecated(ABI_COMPONENT_DEPRECATION_MSG.format("ABIEventParam"))
...
```

Reply:
@reedsa I think the string might need to be generated before passing into the decorator. I factored it out into a method but can you test if without this last change it does work? If not then we may need to generate it before passing it in.